Add a default block icon. #2922

Soean opened the pull request:
## Description
This PR adds the `star-filled` dashicon as default block icon, when no icon is defined.

Resolves #2922

Soean committed:
Add default icon

Soean committed:
Update test

Soean committed:
remove null from block-icon

youknowriad commented on blocks/block-icon/index.js:
```diff
@@ -5,15 +5,20 @@ import { Dashicon } from '@wordpress/components';
 export default function BlockIcon( { icon } ) {
+	if ( ! icon ) {
+		return <Dashicon icon={ 'wordpress' } />;
+	}
```
Wondering if this should be part of the `registerBlock` instead? On the fence. Do you have any thoughts on this?

gziolo replied:
I think it's fine as it is, because [registerBlockType](https://github.com/WordPress/gutenberg/blob/25a2003c7faac2dcdb723161fefd10ef44d147b6/blocks/api/registration.js#L46) does mostly some high-level validations at the moment.

youknowriad replied:
But what if we start using it elsewhere, doing it in `registerBlockType` brings consistency.

gziolo replied:
Both are fine, I was looking only at the currently existing code :)

gziolo commented:
Can we use a different icon as a default one? To test it I removed the icon for Twitter embed. This is what I got:

<img width="279" alt="screen shot 2017-10-11 at 13 38 39" src="https://user-images.githubusercontent.com/699132/31438733-a7aa08d4-ae89-11e7-8fcb-b97f35e4abe0.png">

/cc @jasmussen @karmatosed

jasmussen commented:
I really appreciate this PR, but I agree the default icon probably shouldn't be the WordPress logo. 

Which reminds me there's been zero activity on https://github.com/WordPress/dashicons/issues/203, which also blocks our polishing the embed icons generally. I have made a note to look at drawing both a new default block icon, and providing some better ambiguated embed icons.

karmatosed commented:
I would second that it shouldn't be the W logo. Maybe just a square?

Soean commented:
We should do it in `registerBlockType`, because this does't work.

![fallback](https://user-images.githubusercontent.com/695201/31466672-524d327c-aed8-11e7-8948-d50c07a0eac3.png)

Soean committed:
Move default icon to `registerBlockType`

Soean commented:
I changed the icon to `star-filled` and moved the check into the `registerBlockType` function.

<img width="717" alt="star" src="https://user-images.githubusercontent.com/695201/31468103-a59bc304-aedc-11e7-9737-79c3654f6adc.png">

youknowriad commented:
@jasmussen @karmatosed Are we ok with the star icon? Technically this is good to go.

jasmussen commented:
I've put on my to-do to draw a new default Dashicon, along with some embed icons. I'll just update this too even I've done that. No reason to hold off on this until then, imo, so 👍

youknowriad commented:
Thanks @Soean Nice work
